packages/block-library/src/navigation/index.php

```diff
@@ -637,20 +637,13 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 	}
 
 	$should_load_view_script = ( $has_submenus && ( $attributes['openSubmenusOnClick'] || $attributes['showSubmenuIcon'] ) ) || $is_responsive_menu;
-	$view_js_file            = 'wp-block-navigation-view';
 
-	// If the script already exists, there is no point in removing it from viewScript.
-	if ( ! wp_script_is( $view_js_file ) ) {
-		$script_handles = $block->block_type->view_script_handles;
-
-		// If the script is not needed, and it is still in the `view_script_handles`, remove it.
-		if ( ! $should_load_view_script && in_array( $view_js_file, $script_handles, true ) ) {
-			$block->block_type->view_script_handles = array_diff( $script_handles, array( $view_js_file ) );
-		}
-		// If the script is needed, but it was previously removed, add it again.
-		if ( $should_load_view_script && ! in_array( $view_js_file, $script_handles, true ) ) {
-			$block->block_type->view_script_handles = array_merge( $script_handles, array( $view_js_file ) );
-		}
+	// Load the modules.
+	if ( $should_load_view_script ) {
+		gutenberg_enqueue_module(
+			'@wordpress/block-library/navigation-block',
+			'/wp-content/plugins/gutenberg/build/interactivity/navigation.min.js'
```

Comment:
Do you think we can avoid the file source from the enqueue function? I expect enqueued modules to be registered before hand no? (I know the current scripts API also allows this but it feels weird to me)

Reply:
Sure. I'll do that.

One thing that feels a bit weird to me, though, is to add an entry to the import map for the enqueued modules because they rarely export anything. So I wonder if `register` and `enqueue` should be separate things, one adds entries to the import map and the other adds `<script>` tags. Something like:

- `gutenberg_include_module`: adds the module to the import map
- `gutenberg_enqueue_module`: adds the module to the header using `<script>`

Reply:
I see what you mean, I'm not sure at the moment. I also wonder if there are scripts that could be entry points but also dependencies at the same time.

Reply:
> I also wonder if there are scripts that could be entry points but also dependencies at the same time.

I think it's a possibility, but it's going to be uncommon.